### PR TITLE
Adds check of client's balance before calling monitor

### LINKED ISSUE #30: Check if user has enough tokens deposited
## Description

**As Josh I want the MS to **

check if user has enough tokens deposited before updating the on-chain channel state

**So that it can**

get its reward and be economically viable in the long term.

## Acceptance Criteria

- [x] Before sending the update on chain, check that the UDC has enough balance to pay for the reward
- [x] Add a configurable factor of balance / reward, to stop double spending.

** 3 SP**

### PR DESCRIPTION
**Pull request by Dominik1999**
closes #30

### CODE REVIEW

File: src/monitoring_service/handlers.py [hunk 1]
```diff
@@ -368,9 +372,13 @@ def action_monitoring_triggered_event_handler(event: Event, context: Context) ->
     if channel.update_status:
         last_onchain_nonce = channel.update_status.nonce
 
+    user_address = monitor_request.non_closing_signer
```
Comment (palango) [inline src/monitoring_service/handlers.py]: I'd move that in the call next line

File: tests/monitoring/monitoring_service/test_handlers.py [hunk 2]
```diff
@@ -423,7 +424,7 @@ def test_action_monitoring_triggered_event_handler_does_not_trigger_monitor_call
     assert channel.closing_tx_hash is None
 
 
-def test_action_monitoring_triggered_event_handler_does_trigger_monitor_call(  # noqa
+def test_action_monitoring_triggered_event_handler_with_suff_balance_does_trigger_monitor_call(  # noqa
```
Comment (palango) [inline tests/monitoring/monitoring_service/test_handlers.py]: can you spell out sufficient here

File: tests/monitoring/monitoring_service/test_handlers.py [hunk 3]
```diff
@@ -455,22 +456,80 @@ def test_action_monitoring_triggered_event_handler_does_trigger_monitor_call(  #
...
+    context: Context,
+):
+    """ Tests that `monitor` is called when the ActionMonitoringTriggeredEvent is triggered
+
+    Also a test for https://github.com/raiden-network/raiden-services/issues/29 , as the MR
+    is sent after the channel has been closed.
+    """
+    context = setup_state_with_closed_channel(context)
+
+    event_new_bp = ReceiveMonitoringNewBalanceProofEvent(
```
Comment (palango) [inline tests/monitoring/monitoring_service/test_handlers.py]: This thing can be removed, there is another test for this.

File: tests/monitoring/monitoring_service/test_handlers.py [hunk 4]
```diff
@@ -491,6 +550,8 @@ def test_mr_available_before_channel_triggers_monitor_call(
         non_closing_participant=DEFAULT_PARTICIPANT2,
     )
 
+    user_address = DEFAULT_PARTICIPANT2
```
Comment (palango) [inline tests/monitoring/monitoring_service/test_handlers.py]: I'd also inline this.

File: tests/monitoring/monitoring_service/test_handlers.py [hunk 5]
```diff
@@ -455,22 +456,80 @@ def test_action_monitoring_triggered_event_handler_does_trigger_monitor_call(  #
...
+
+    channel = context.db.get_channel(
+        trigger_event.token_network_address,
+        trigger_event.channel_identifier,
+    )
+    assert channel
+    assert channel.update_status is not None
+    assert channel.closing_tx_hash is None
+
+    user_address = DEFAULT_PARTICIPANT2
```
Comment (palango) [inline tests/monitoring/monitoring_service/test_handlers.py]: same here

File: tests/monitoring/monitoring_service/test_handlers.py [hunk 6]
```diff
@@ -95,7 +95,7 @@ def get_signed_monitor_request(
     )
     monitor_request = UnsignedMonitorRequest.from_balance_proof(
         bp,
-        reward_amount=0,
+        reward_amount=10,
```
Comment (palango) [inline tests/monitoring/monitoring_service/test_handlers.py]: Can you make this a parameter of the function as well?

File: src/monitoring_service/handlers.py (hunk 1 above)
Comment (Dominik1999) [reply src/monitoring_service/handlers.py]: Yes that would be better, but then the line is too long. The function call is a monster!

File: tests/monitoring/monitoring_service/test_handlers.py (hunk 2 above)
Comment (Dominik1999) [reply tests/monitoring/monitoring_service/test_handlers.py]: yes sure that makes sense

File: tests/monitoring/monitoring_service/test_handlers.py (hunk 4 above)
Comment (Dominik1999) [reply tests/monitoring/monitoring_service/test_handlers.py]: MyPy says no ... 
```
tests/monitoring/monitoring_service/test_handlers.py:473:100: E501 line too long (105 > 99 characters)
tests/monitoring/monitoring_service/test_handlers.py:525:100: E501 line too long (104 > 99 characters)
```

File: tests/monitoring/monitoring_service/test_handlers.py (hunk 6 above)
Comment (Dominik1999) [reply tests/monitoring/monitoring_service/test_handlers.py]: yes, sure we talked about that

Comment (Dominik1999) [reply tests/monitoring/monitoring_service/test_handlers.py]: done

File: tests/monitoring/monitoring_service/test_handlers.py (hunk 3 above)
Comment (Dominik1999) [reply tests/monitoring/monitoring_service/test_handlers.py]: OK I removed it

File: src/monitoring_service/handlers.py (hunk 1 above)
Comment (palango) [reply src/monitoring_service/handlers.py]: Can you inline this as well

File: tests/monitoring/monitoring_service/test_handlers.py [hunk 7]
```diff
@@ -423,7 +426,7 @@ def test_action_monitoring_triggered_event_handler_does_not_trigger_monitor_call
     assert channel.closing_tx_hash is None
 
 
-def test_action_monitoring_triggered_event_handler_does_trigger_monitor_call(  # noqa
+def test_action_monitoring_triggered_event_handler_with_sufficient_balance_does_trigger_monitor_call(  # noqa
     context: Context,
 ):
     """ Tests that `monitor` is called when the ActionMonitoringTriggeredEvent is triggered
```
Comment (palango) [inline tests/monitoring/monitoring_service/test_handlers.py]: Please also update the docstrings for this and next test.

Comment (Dominik1999) [reply tests/monitoring/monitoring_service/test_handlers.py]: good point, done

**Pull request merged**